games-fps/serious-sam-tse: switch to fork for working package

tx00100xt opened the pull request:
Bug: https://bugs.gentoo.org/854567

mm1ke commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,76 @@
+
+DEPEND="games-fps/serioussam"
+
+S=${WORKDIR}
```
you already declared `S` a few lines before here - this can be removed ;)

mm1ke commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,76 @@
+
+src_unpack() {
+	mkdir Levels Mods
+	cat "${DISTDIR}/${PATCH_PREFIX}.tar.xz" > "${PATCH_PREFIX}.tar.xz"
```
missing `|| die` for both `mkdir` and `cat`

mm1ke commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
`EAPI=8`, afaics both `cdrom` and `unpacker` alread support `EAPI=8`

mm1ke commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,76 @@
+GN="serioussam"
+
+DESCRIPTION="Croteam's Serious Sam Classic The First Encounter ... the data files"
+HOMEPAGE="http://www.croteam.com/
```
the homepage supports `https`
```
HOMEPAGE="https://www.croteam.com/"
```

mm1ke commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,76 @@
+	rm -rf "${D}/${dir}"/Scripts || die "failed to removed stuff"
+	rm -rf "${D}/${dir}"/VirtualTrees || die "failed to removed stuff"
+	rm -f  "${D}/${dir}"/Help/ShellSymbols.txt || die "failed to removed stuff"
+	mkdir "${D}/${dir}/Scripts"
```
missing `|| die`

mm1ke commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,76 @@
+	elog "    look at:"
+	elog "  https://github.com/tx00100xt/SeriousSamClassic-VK/wiki"
+	elog "   For information about of the game"
+
+	echo
```
I think these two lines could be removed.

mm1ke commented:
I've made some comments too the first ebuild, but some/most of them apply to the others too. In general some more suggestions:
* please add one commit per packages
* don't remove the `games-fps/serious-sam-tse` directly. This should be masked first and if `games-fps/serious-sam-tse` would collide with the new packages we would need to make blockers too.
* I hope the package were tested on both `~amd64` and `~x86`? Otherwise i guess `~amd64` should be fine.

Thx for looking at this package. I was thinking to have a look at it too but without having the game itself it would be a bit difficult to test.

tx00100xt commented:
Thanks for watching.
There are no conflicts with package games-fps/serious-sam-tse.
I put "die" everywhere. EAPI changed to 8 in ebuilds serioussam-tfe-data-1.5-r1.ebuild and serioussam-tse-data-1.7-r1.ebuild
The game runs on x86. But I only tested ebuilds on amd64. Therefore, I left only amd64.
For Croteam, the home page has been corrected. Duplicate ${WORKDIR} removed, echo also removed it at the end.
Сhecked all the builds for functionality. I always have ISO images of CD games at hand.

tx00100xt commented on games-fps/serious-sam-tse/serious-sam-tse-1_beta1-r1.ebuild:
```diff
@@ -0,0 +1,137 @@
+DESCRIPTION="Serious Sam: The Second Encounter"
+HOMEPAGE="http://www.croteam.com/
+	http://www.seriouszone.com/"
+SRC_URI="http://icculus.org/betas/ssam/${MY_PN}-${MY_PV}.sh.bin"
```
@ulm 
I made all the changes as you asked.
Including those listed above.
I deleted it http://www.seriouszone.com/ because the forum closed a few years ago.

tx00100xt commented:
@ulm
After the changes to ebuild serious-sam-tse-1_beta1-r1.ebuild, I checked the building.
The game even started. But the binaries in it are very old.
After the start, there were graphical bugs and the game is going too fast because it does not know how to work with TSC p-state.

ulm commented:
> @ulm After the changes to ebuild serious-sam-tse-1_beta1-r1.ebuild, I checked the building. The game even started. But the binaries in it are very old. After the start, there were graphical bugs and the game is going too fast because it does not know how to work with TSC p-state.
> 
> **upd.** Graphic artifacts removed by changing the settings. But the game speed exceeds the required one several times.

Then I'd suggest then that you leave the `games-fps/serious-sam-tse` package alone altogether, i.e. drop the first two commits from the series.

It even looks like a removal candidate. Could you file a bug for this (and include the findings you've posted above)?

tx00100xt commented:
> It even looks like a removal candidate. Could you file a bug for this (and include the findings you've posted above)?

I was reporting bugs in the package to https://bugs.gentoo.org/854567
Do I need to add anything else there?
I can add a comment there at the end. 
> Then I'd suggest then that you leave the games-fps/serious-sam-tse package alone altogether, i.e. drop the first two commits from the series.
>
I'll do it now.

tx00100xt commented:
>It even looks like a removal candidate. Could you file a bug for this (and include the findings you've posted above)?
>
@ulm
I added it as you requested.
https://bugs.gentoo.org/854567#c22

tx00100xt commented:
@mm1ke, @ulm
Hello!
I have an old computer (FX-4100 with Radeon HD3000 graphics built into the chipset).
I installed Gentoo on it from the latest stage3-i686 to test the ebuild. When checking, there was a warning about TEXTREL due to the use of assembler (usual for 32-bit architecture with assembler). I made a change to the source code to use only c/c++ for i686. After this there were no warnings to QA Notice. I made a new tag 1.10.6c, now ebuild uses it. Additionally, I made changes to CMakeLists.txt so that there were no unnecessary patches and added man pages. After checking the i686 I added ~x86 to the ebuild. If anything else needs to be fixed, please write.

tx00100xt commented:
I removed unnecessary commits. I left only three, one commit per package.

mm1ke commented on games-fps/serioussam/serioussam-1.10.6.ebuild:
```diff
@@ -0,0 +1,151 @@
+
+	einfo "Setting build type Release..."
+	CMAKE_BUILD_TYPE="Release"
+	local mycmakeargs=(
+		-DUSE_SYSTEM_INSTALL=ON
+	)
+	if use vulkan; then
+		mycmakeargs+=(
+			-DUSE_SYSTEM_VULKAN=ON
+		)
+	else
+		mycmakeargs+=(
+			-DUSE_SYSTEM_VULKAN=OFF
+		)
+	fi
+	if use x86; then
+		mycmakeargs+=(
+			-DUSE_ASM=OFF
+		)
+	fi
```
i think you could simplify this a bit:
```suggestion
	local mycmakeargs=(
		-DUSE_SYSTEM_INSTALL=ON
		-DUSE_SYSTEM_VULKAN=$(usex vulkan ON OFF)
		-DUSE_ASM=$(usex x86 OFF ON)
	)
```
Haven't tested this so please check if it works .

tx00100xt replied:
Works. Tested x86 and amd64.
Everything works correctly.

mm1ke commented on games-fps/serioussam/serioussam-1.10.6.ebuild:
```diff
@@ -0,0 +1,151 @@
+			-DUSE_ASM=OFF
+		)
+	fi
+	BUILD_DIR="${BUILD_TMP}"
```
first you declare 
```BUILD_TMP=${BUILD_DIR}```
and here you set:
```BUILD_DIR="${BUILD_TMP}"```

This seems to be a bit redundant? Could it be removed at all? I don't even know where this variable comes from - it doesn't seems to be a common ebuild variable?

mm1ke commented on games-fps/serioussam/serioussam-1.10.6.ebuild:
```diff
@@ -0,0 +1,151 @@
+	cmake_src_configure
+}
+
+src_compile() {
+	einfo "Compiling standart game..."
+	cmake_src_compile
+}
```
I _think_ this could be removed completely? The `einfo` is not really helpful and `cmake_src_compile` would be called anyway.

tx00100xt replied:
Deleted.

mm1ke commented on games-fps/serioussam/serioussam-1.10.6.ebuild:
```diff
@@ -0,0 +1,151 @@
+	media-gfx/imagemagick
+"
+
+S="${WORKDIR}/SeriousSamClassic-VK-${PV}c"
+MY_CONTENT1="${WORKDIR}/SeriousSamClassic-VK-${PV}c/${MY_PN1}"
+MY_CONTENT2="${WORKDIR}/SeriousSamClassic-VK-${PV}c/${MY_PN2}"
```
This probably should be moved below `SRC_URI`, at least `S` belongs there.

tx00100xt commented on games-fps/serioussam-tse-data/serioussam-tse-data-1.7-r1.ebuild:
```diff
@@ -0,0 +1,73 @@
+	mv "${WORKDIR}"/*.gro "${D}/${dir}" || die "failed to moved patch 1.07"
+	mv "${D}/${dir}"/Scripts/PersistentSymbols.ini "${WORKDIR}" || die "failed to moved PersistentSymbols.ini"
+
+	rm -rf "${D}/${dir}"/Bin || die "failed to removed stuff"
+	rm -rf "${D}/${dir}"/Controls || die "failed to removed stuff"
+	rm -rf "${D}/${dir}"/Data || die "failed to removed stuff"
+	rm -rf "${D}/${dir}"/Demos || die "failed to removed stuff"
+	rm -rf "${D}/${dir}"/Mods || die "failed to removed stuff"
+	rm -rf "${D}/${dir}"/Players || die "failed to removed stuff"
+	rm -rf "${D}/${dir}"/Scripts || die "failed to removed stuff"
+	rm -rf "${D}/${dir}"/VirtualTrees || die "failed to removed stuff"
+	rm -f  "${D}/${dir}"/Help/ShellSymbols.txt || die "failed to removed stuff"
+	rm -f  "${D}/${dir}"/ModEXT.txt || die "failed to removed stuff"
+	mkdir "${D}/${dir}/Scripts" || die "failed create Scripts dir"
+	mv "${WORKDIR}"/PersistentSymbols.ini "${D}/${dir}/Scripts" || die "failed to moved PersistentSymbols.ini"
+
+	rm -rf "${D}/${dir}"/Locales || die "failed to removed stuff"
```
> However, if there are a lot of files i think it fine to first remove and create it again.
>
There are old configuration files on the disk. New ones are taken from the repo in serioussam-1.10.6.ebuild

tx00100xt commented:
I made all the necessary changes. Tested all three ebuilds.

tx00100xt commented:
I've done everything. If you need anything else please write.

mm1ke commented:
@tx00100xt 

I made some more comments.
Also you quite often use `|| die "Failed to removed temp stuff"`, which i would just change to `|| die` or at least `|| die "failed to remove file X` to make it a bit more useful.
Also the `pkg_postinst` is quite huge and i'm wondering if it wouldn't be better to put that into `DOC_CONTENTS`. Also regarding this additional information - wouldn't users need to remove them manually too when uninstalling `serioussam`. I would also add some `pkg_postrm` to let users know to remove these files in case they uninstall `serioussam`.

tx00100xt commented:
@mm1ke
I'll do it now.

tx00100xt commented:
- The manpage is detailed. 
  Can leave it and the links to the repo in pkg_postinst.
- pkg_postrm has been added.
  Now the user does not need to delete anything manually.
- "die" made it more informative

tx00100xt commented:
I can replace pkg_postrm with this one, if necessary:
>pkg_postrm() {
>      elog "     For complete uninstallation, remove directories:"
>      elog "        /usr/share/serioussam"
>      elog "        /usr/share/serioussamse"
>      elog "
>}
>
I don’t know what’s better, rm or just inform

ulm commented:
>     * pkg_postrm has been added.
>       It says what needs to be removed for complete uninstallation

Probably I am missing something, but why would there be files in `/usr/share` that aren't removed when unmerging the package?

tx00100xt commented:
@ulm
I corrected and rechecked the build.
Now, if the user manually added game data, instead of the serioussam-tfe-data and serioussam-tse-data ebuilds, these will also be deleted. All that remains is to clean up your home directory (~/.local/share/Serious-Engine).

mm1ke commented:
> @mm1ke When using serious sam-tfe-data and serious sam-tse-data everything should be tracked. In case of manually adding files, I added rm -fr in pkg_postrm. If you can suggest a more correct solution, then please write
> 
> When using ebuilds serioussam-tfe-data and serioussam-tse-data, you do not need to copy anything additionally, only if you install one serious package

Ok, i think the correct solutions would be to add an `pkg_postinst()` to point to the upstream wiki for additional information, something like: 
```
For additional information about this `serioussam` fork visit:
https://github.com/tx00100xt/SeriousSamClassic-VK
```
but keep it short, there's no need to add the whole manual installation to the ebuild, especially since it works anyway without it. In this case the `pkg_postrm` could be dropped too.

mm1ke commented:
BTW, forgot to menitioned.
`games-fps/serioussam` would also need to depend on `games-fps/serioussam-tse-data` and `games-fps/serioussam-tfe-data`

tx00100xt commented:
Ready. I checked the package build twice. There are no untracked files.

ulm commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,62 @@
+# Game name
+GN="serioussam"
+
+DESCRIPTION="Croteam's Serious Sam Classic The First Encounter ... the data files"
+HOMEPAGE="https://www.croteam.com/
+	https://store.steampowered.com/app/41050/Serious_Sam_Classic_The_First_Encounter/"
+SRC_URI="https://github.com/tx00100xt/serioussam-mods/raw/main/Patches/${PATCH_PREFIX}.tar.xz"
+
+S="${WORKDIR}"
```
This is supposed to be a single variables block, so no blank line between `SRC_URI` and `S` please:
```suggestion
DESCRIPTION="Croteam's Serious Sam Classic The First Encounter ... the data files"
HOMEPAGE="https://www.croteam.com/
	https://store.steampowered.com/app/41050/Serious_Sam_Classic_The_First_Encounter/"
SRC_URI="https://github.com/tx00100xt/serioussam-mods/raw/main/Patches/${PATCH_PREFIX}.tar.xz"
S="${WORKDIR}"
```

ulm commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,62 @@
+
+src_unpack() {
+	mkdir Levels Mods || die "failed create dirs"
+	cat "${DISTDIR}/${PATCH_PREFIX}.tar.xz" > "${PATCH_PREFIX}.tar.xz" || die "failed to copy patch 1.05"
```
Please keep lines shorter than 80 chars if possible.
```suggestion
	cat "${DISTDIR}/${PATCH_PREFIX}.tar.xz" > "${PATCH_PREFIX}.tar.xz" \
		|| die "failed to copy patch 1.05"
```

(Also in several following instances.)

ulm commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,62 @@
+	mv "${D}/${dir}"/Scripts/PersistentSymbols.ini "${WORKDIR}" || die "failed to moved PersistentSymbols.ini"
+
+	rm -rf \
+		"${D}/${dir}"/{Bin,Controls,Data,Demos,Mods,Players,Scripts,VirtualTrees,Locales,Help/ShellSymbols.txt} \
```
serioussam-tse-data also removes `ModEXT.txt`, is it correct that it isn't listed here?

ulm commented on games-fps/serioussam-tse-data/serioussam-tse-data-1.7-r1.ebuild:
```diff
@@ -0,0 +1,62 @@
+DESCRIPTION="Croteam's Serious Sam Classic The Second Encounter ... the data files"
+HOMEPAGE="https://www.croteam.com/
+	https://store.steampowered.com/app/41060/Serious_Sam_Classic_The_Second_Encounter/"
+SRC_URI="https://github.com/tx00100xt/serioussam-mods/raw/main/Patches/${PATCH_PREFIX}.tar.xz"
+
+S="${WORKDIR}"
```
See above, no blank line here.

ulm commented on games-fps/serioussam/serioussam-1.10.6.ebuild:
```diff
@@ -0,0 +1,102 @@
+
+DESCRIPTION="Linux port of Serious Sam Classic with Vulkan support"
+HOMEPAGE="https://github.com/tx00100xt/SeriousSamClassic-VK"
+SRC_URI="https://github.com/tx00100xt/SeriousSamClassic-VK/archive/refs/tags/${PV}c.tar.gz -> ${P}c.tar.gz"
+
+S="${WORKDIR}/SeriousSamClassic-VK-${PV}c"
+MY_CONTENT1="${WORKDIR}/SeriousSamClassic-VK-${PV}c/${MY_PN1}"
```
```suggestion
SRC_URI="https://github.com/tx00100xt/SeriousSamClassic-VK/archive/refs/tags/${PV}c.tar.gz -> ${P}c.tar.gz"
S="${WORKDIR}/SeriousSamClassic-VK-${PV}c"

MY_CONTENT1="${WORKDIR}/SeriousSamClassic-VK-${PV}c/${MY_PN1}"
```

ulm commented on games-fps/serioussam/serioussam-1.10.6.ebuild:
```diff
@@ -0,0 +1,102 @@
+	elog "        https://github.com/tx00100xt/SeriousSamClassic-VK/wiki"
+	elog "        https://github.com/tx00100xt/SeriousSamClassic-VK/README.md"
+	elog "     For information on the first launch of the game"
+	elog ""
```
Drop the empty `elog` command.

ulm reviewed (changes requested):
Overall this looks good to me, so I think it's time to address some style/cosmetic issues. :smile:

tx00100xt commented:
> Overall this looks good to me, so I think it's time to address some style/cosmetic issues. 😄

I have read all the comments. I'll fix it now

tx00100xt commented:
@ulm
Fixed. And checked the build. If you need anything else, please write.

tx00100xt commented on games-fps/serioussam/serioussam-1.10.6.ebuild:
```diff
@@ -0,0 +1,109 @@
+MY_CONTENT1="${WORKDIR}/SeriousSamClassic-VK-${PV}c/${MY_PN1}"
+MY_CONTENT2="${WORKDIR}/SeriousSamClassic-VK-${PV}c/${MY_PN2}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+RESTRICT="bindist mirror"
```
> Is GPL-2 the only license of the main serioussam package?

* Serious Engine is licensed under the GNU GPL v2.
* amp11lib is licensed under the GNU GPL v2.

Some of the code included with the engine sources is not licensed under the GNU GPL v2:

* zlib (located in `Sources/Engine/zlib`) by Jean-loup Gailly and Mark Adler
* LightWave SDK (located in `Sources/LWSkaExporter/SDK`) by NewTek Inc.
* libogg/libvorbis (located in `Sources/libogg` and `Sources/libvorbis`) by Xiph.Org Foundation

ulm replied:
> Some of the code included with the engine sources is not licensed under the GNU GPL v2:
> 
>     * zlib (located in `Sources/Engine/zlib`) by Jean-loup Gailly and Mark Adler
> 
>     * LightWave SDK (located in `Sources/LWSkaExporter/SDK`) by NewTek Inc.
> 
>     * libogg/libvorbis (located in `Sources/libogg` and `Sources/libvorbis`) by Xiph.Org Foundation

OK, so it should be `LICENSE="GPL-2 BSD ZLIB"` with `BSD` for libogg/libvorbis and `ZLIB` for zlib.

What is the license of the files in `Sources/LWSkaExporter/SDK`? Looks like these are trivial header files?

tx00100xt replied:
https://github.com/tx00100xt/SeriousSamClassic/blob/main/SamTFE/Sources/LWSkaExporter/SDK/lwanimlod.h

>/*
 * LWSDK Header File
 * Copyright 1999,  NewTek, Inc.
 *
 * LWANIMLOD.H -- LightWave Animation Loaders
 */

tx00100xt replied:
These headers are used only when building for Windows

tx00100xt replied:
I corrected all three ebuilds.

thesamesam commented on games-fps/serioussam-tfe-data/Manifest:
```diff
@@ -0,0 +1 @@
+DIST Patch_1.05_FE.tar.xz 1221836 BLAKE2B 0ae7103151ee42f3c1ff88afd2d77037e0c3e5c49c19adf213068f87eed524c7aa6552d020c3337f5457847f167c1a2e4ca1fda7572e96b9630fb2dbfd3e6b0f SHA512 18c5a5f66e1fbe3864eca3274d20e525f50c223507e0c1c793a642aed66edfeb723cf3d4ffc583afa4d9a10388cbed14af7035bf6a1b9c214dac79c5eb1d2548
```
Should we rename this distfile? It feels at risk of collisions (too generic).

thesamesam commented on games-fps/serioussam-tfe-data/serioussam-tfe-data-1.5-r1.ebuild:
```diff
@@ -0,0 +1,70 @@
+	doins -r "${CDROM_ROOT}"/Install/*
+
+	mv "${WORKDIR}"/*.gro "${D}/${dir}" || die "failed to moved patch 1.05"
+	mv "${D}/${dir}"/Scripts/PersistentSymbols.ini "${WORKDIR}" \
```
This should be "${ED}", not "${D}", as ${dir} does not include ${EPREFIX}. Same below too.

Or just prepend it to dir.

thesamesam commented on games-fps/serioussam-tse-data/Manifest:
```diff
@@ -0,0 +1 @@
+DIST Patch_1.07_SE.tar.xz 33172 BLAKE2B bc185d0b504e7aa01862a57ba97691881e19d0031ac88187ec80fb52d8bd758c0dac6b932a333fa4f9df09264232ab87e97875a8ae7a58cd0816153a082ffc45 SHA512 1c0e2ac2503584d26ac7fc59f72b0dcb6903bfc4897acc5732b426fe691928c1f49839b3a893809e635e178e447aa72fb9969179806a1703d412c4ea321ca637
```
Possibly rename?

tx00100xt replied:
Yes.
seriossamfe-patch_1.05_FE.tar.xz
seriossamse-patch_1.07_SE.tar.xz
Are these names suitable?

thesamesam replied:
Sounds good!

ulm commented:
Merged with the small changes suggested by @thesamesam.

Thank you!

tx00100xt commented:
Patch file names have been changed:
https://github.com/tx00100xt/serioussam-mods/tree/main/Patches

Thank you all for participating.

ulm commented:
> Patch file names have been changed: https://github.com/tx00100xt/serioussam-mods/tree/main/Patches

Typo? Should be serio**u**ssamfe-patch... I guess.

tx00100xt commented:
I'll fix it now

ulm commented:
`SRC_URI` of ebuilds updated, so should be all set now.

tx00100xt commented:
I also added a copy with the old names.
